blackbird_python/blackbird/tests/test_listener.py

```diff
@@ -346,6 +347,313 @@ def test_target_arg(self, parse_input):
+            f.write(template1)
+
+        template2 = textwrap.dedent(
+            """
+            name Chip15
+            version 0.0
+            include "{}"
+            S2gate({{sq}}) | [0, 1]
+            MachZehnder(theta={{theta}}, phi={{phi}}) | [1, 2]
+            """
+        ).format(filename1)
+
+        filename2 = tmpdir.join("chip16.xbb")
+
+        with open(filename2, "w") as f:
+            f.write(template2)
+
+        test_include = textwrap.dedent(
+            """
+            name test_include
+            version 0.0
+            include "{}"
+            include "{}"
+            Chip15(sq=0.5, theta=0.54, phi=0.1) | [0, 2, 1]
+            """
+        ).format(filename2, filename1)
+
+        bb = parse_input(test_include, cwd=tmpdir)
+
+        expected = [
+            {"op": "S2gate", "args": [0.5], "kwargs": {}, "modes": [0, 2]},
+            {"op": "BSgate", "args": [0.54, np.pi / 2], "kwargs": {}, "modes": [2, 1]},
+            {"op": "Rgate", "args": [0.1], "kwargs": {}, "modes": [2]},
+        ]
+
+        assert bb.operations == expected
+
+    @pytest.mark.skipif(sys.version_info < (3, 6), reason="tmpdir fixture requires Python >=3.6")
+    def test_mismatched_modes(self, parse_input, tmpdir):
+        """Test exception raised if modes do not match between
+        applied subroutine and the defined template"""
+        program = textwrap.dedent(
+            """
+            name CustomOperation
+            version 0.0
+            float alpha = 0.3423
+            Coherent(alpha, sqrt(pi)) | 0
+            MeasureFock() | 0
+            """
+        )
+
+        filename = tmpdir.join("test.xbb")
+
+        with open(filename, "w") as f:
+            f.write(program)
+
+        test_include = textwrap.dedent(
+            """
+            name test_include
+            version 0.0
+            include "{}"
+            CustomOperation | [0, 1]
+            """
+        ).format(filename)
+
+        with pytest.raises(
+            ValueError, match="CustomOperation acts on 1 modes, but 2 modes provided"
+        ):
+            bb = parse_input(test_include, cwd=tmpdir)
+
+    @pytest.mark.skipif(sys.version_info < (3, 6), reason="tmpdir fixture requires Python >=3.6")
+    def test_too_many_arguments(self, parse_input, tmpdir):
+        """Test exception raised if custom subroutine does
+        not accept arguments, but arguments are passed:"""
+        program = textwrap.dedent(
+            """
+            name CustomOperation
+            version 0.0
+            float alpha = 0.3423
+            Coherent(alpha, sqrt(pi)) | 0
+            MeasureFock() | 0
+            """
+        )
+
+        filename = tmpdir.join("test.xbb")
+
+        with open(filename, "w") as f:
+            f.write(program)
+
+        test_include = textwrap.dedent(
+            """
+            name test_include
+            version 0.0
+            include "{}"
+            CustomOperation() | 0
```

Comment:
I don't see where the argument is passed 🤔
There are brackets, but no argument

Reply:
Exactly what I wanted 🙂 This test tests that that fails:

```python
with pytest.raises(ValueError, match="CustomOperation does not accept arguments"):
    bb = parse_input(test_include, cwd=tmpdir)
```